### ISSUE TITLE
Format of percentage not as expected

### ISSUE THREAD

**Issue by kieranjmartin** [Roche/rtables#6]
In master branch, a percentage format doesnt give .0 if a value rounds to 1/0.

E.G.
`   rcell(c(10, 0.1101), format = "xx (xx.x%)")`

returns
`10 (11%)`

rather than

`10 (11.0%)`

**Comment by waddella** [comment]
what do you think of adding an additional format `xx (xx..x%)` where `..` would enforce the printing of trailing zeros?

**Comment by kieranjmartin** [comment]
I don't know: I feel like trailing zeros is whats expected when you are using a particular level of significance, so I'd rather have the default behaviour act like that. Certainly you could give both as an option though.

**Comment by waddella** [comment]
Hi @kieranjmartin,

We have now on the devel branch the `sprinf_format` function where

```r
rtable(
  header = c("format label", "sprintf_format"),
  rrow("", rcell(2, "xx.xx"), rcell(2, sprintf_format("%.2f")))
)
```

yields

```
         format label         sprintf_format
--------------------------------------------
              2                    2.00     
```

however we still need to decide whether `xx.xx` should give `2.00`.

**Comment by kieranjmartin** [comment]
Well, my feeling is that "xx.xx" gives the expectation of always having trailing digits. Looking at implementations of formats Excel follows this paradigm, which I suspect is what most users would be familiar with

**Comment by waddella** [comment]
I agree, I'll discuss this with some other users, but if no good reasons come up to leave it as is then I'll change that in the next release.

Thanks for bringing that up.

**Comment by thomas-neitmann** [comment]
I'd like to restart this discussion as this is really bothering me and my team who use {rtables} for reporting an IMC. Just as @kieranjmartin mentioned if I have a format like `xx (xx.x%)` then I **always** want to have a trailing digits after the decimal dot. In fact this point has been raised by our stakeholders in clinical and safety science. Thus, we really need to take care of it and I hope to see a fix for this in the next release. Btw, more or less the same issue has been raised [here](https://github.roche.com/NEST/rtables/issues/43).

I've went ahead and *started* to fix these issues in #128. @waddella @gmbecker @danielinteractive Please let me know if this looks good to you. If so, I would continue fixing the remaining formats. This is a much needed fix and I'd appreciate a speedy response from you. Thanks!

**Comment by danielinteractive** [comment]
Thanks @thomas-neitmann, I also have seen this before, so +1 on this effort. Generally the PR looks like the right thing, but could you instead please request the pull on public github into the gabe_table_tree branch? since that is our ongoing development branch. Thank you

**Comment by thomas-neitmann** [comment]
Will do, thanks @danielinteractive!

**Comment by anajens** [comment]
@waddella @gmbecker it would be great if this issue can be prioritized for the next release.

**Comment by waddella** [comment]
@gmbecker as discussed today in the meeting, lets use `sprintf` on the rounded numbers using `round` in order to keep using the R rounding rules.

**Comment by gmbecker** [comment]
This is addressed fully in #272 and can be closed when that is merged in.
